### PR TITLE
Support finetuning with the AxoNN backend

### PR DESCRIPTION
**Pull request by siddharth9820**
Changes apart from AxoNN - 

- [x] multi-node support
- [x] lower lr of optimizer (the default is too high for finetuning)
- [x] add gradient clipping

### CODE REVIEW

**Comment by siddharth9820** [comment]
Evidence for correctness - 

1. Identical loss curves with FSDP in fp-16
image
